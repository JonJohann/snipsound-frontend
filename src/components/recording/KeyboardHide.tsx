import React from 'react';
import { TouchableOpacity, Keyboard } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import { styles } from '../../styles/styles'

// Button for hiding the keyboard
export default function KeyboardHide(props: any) {
    
    return (
        <TouchableOpacity style={[styles.symbols, { marginTop: 15, marginRight: -1, marginLeft: 2}]} onPress={() => {Keyboard.dismiss(); props.setVisibleKeyboard(false)}}>
            <MaterialIcons style={{opacity: props.visibleKeyboard ? 1 : 0}} name="keyboard-hide" size={30} color="black" />
        </TouchableOpacity>
    )
}