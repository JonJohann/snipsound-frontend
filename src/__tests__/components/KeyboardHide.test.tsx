import React from 'react';
import renderer from 'react-test-renderer';
import { Keyboard } from 'react-native';
import KeyboardHide from '../../components/recording/KeyboardHide';
import Enzyme, { shallow }from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })

describe('<KeyboardHide />', () => {
    const setVisibleKeyboard = jest.fn();
    const visibleKeyboard = false;

    const props = {
        setVisibleKeyboard,
        visibleKeyboard,
    }
    jest.mock('react-native');
    
    jest.mock('react-native', () => ({
        NativeModules: {
          RNPasscodeStatus: {
            supported: jest.fn(),
            status: jest.fn(),
            get: jest.fn(),
          }
        },
        StyleSheet: {
          create: () => ({})
        },
        Platform: {
          OS: jest.fn(() => 'ios'),
          version: jest.fn(() => 25),
        },
      }));

    const mockKeyboardDismiss = Keyboard.dismiss as jest.MockedFunction<typeof Keyboard.dismiss>;
    it('calls setVisibleKeyboard on click', () => {
        const _component = shallow(<KeyboardHide {...props}/>)
        _component.simulate('press')
        expect(setVisibleKeyboard).toHaveBeenCalledTimes(1)
    })

    it('renders correctly', () => {
        const _component = renderer.create(<KeyboardHide {...props}/>).toJSON();
        expect(_component).toMatchSnapshot();
    });

});