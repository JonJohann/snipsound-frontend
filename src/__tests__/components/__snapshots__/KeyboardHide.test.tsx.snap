// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeyboardHide /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "margin": 10,
      "marginLeft": 2,
      "marginRight": -1,
      "marginTop": 15,
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
